lib/presentation/recherche/recherche_home_page_view_model.dart

```diff
@@ -18,9 +20,16 @@ class RechercheHomePageViewModel extends Equatable {
         OffreType.immersion,
         if (isCej) OffreType.serviceCivique,
       ],
+      shouldShowOnboarding: _shouldShowOnboarding(store),
     );
   }
 
   @override
-  List<Object?> get props => [offreTypes];
+  List<Object?> get props => [offreTypes, shouldShowOnboarding];
+}
+
+bool _shouldShowOnboarding(Store<AppState> store) {
```

Comment:
Je vois que tu as fait ces méthodes privées 4 fois. Ca vaut le coup tu penses de mettre directement dans la sealed class OnboardingState `bool get showAccueilOnboarding {
return this is OnboardingSuccessState && (this as OnboardingSuccessState).result.showAccueilOnboarding == true;
}` (qui serait d'ailleurs `bool get showAccueilOnboarding {
return this is OnboardingSuccessState && (this as OnboardingSuccessState).result.showAccueilOnboarding;
}` si tu prends mon autre retour sur mettre les bool en non nullable)